controllers/controllers/resource/fetcher.go

```diff
@@ -302,11 +306,11 @@ func (r *capiResourceFetcher) FetchAppliedSpec(ctx context.Context, cs *anywhere
 }
 
 func (r *capiResourceFetcher) ExistingVSphereDatacenterConfig(ctx context.Context, cs *anywherev1.Cluster) (*anywherev1.VSphereDatacenterConfig, error) {
-	vsMachineTemplate, err := r.VSphereWorkerMachineTemplate(ctx, cs)
+	vsMachineTemplates, err := r.VSphereWorkerMachineTemplates(ctx, cs)
 	if err != nil {
 		return nil, err
 	}
-	return MapMachineTemplateToVSphereDatacenterConfigSpec(vsMachineTemplate)
+	return MapMachineTemplateToVSphereDatacenterConfigSpec(&vsMachineTemplates[0])
```

Comment:
I would feel better accessing the first if we know it's a template that we maintain

Reply:
I figured we could still use the first since all machine deployments will have the same datacenter config

Reply:
Yea but what if from the machine deployment call we pull in a template the user brought in, which would then break our controller assumptions? I was mainly worried about that edge case.